Sources/AppStoreConnectCLI/Commands/Certificates/ListDownloadCertificates.swift

```
@@ -0,0 +1,93 @@
+            discussion: "The `-` prefix indicates descending order."
+        )
+    )
+    var sort: Certificates.Sort?
+
+    @Option(help: "The type of certificate to create \(CertificateType.allCases).")
+    var filterType: CertificateType?
+
+    @Option(help: "The certificate’s display name. (eg. Mac Installer Distribution: TeamName)")
+    var filterDisplayName: String?
+
+    @Option(help: "Limit the number of resources (maximum 200).")
+    var limit: Int?
+
+    @Option(help: "The file download path. (eg. ~/Documents)")
+    var downloadPath: String?
+
+    enum CommandError: LocalizedError {
+        case invalidPath(String)
+        case invalidContent
+
+        var errorDescription: String? {
+            switch self {
+            case .invalidPath(let path):
+                return "Download failed, please check the path '\(path)' you input and try again"
+            case .invalidContent:
+                return "The certificate in the response doesn't have a proper content"
+            }
+        }
+    }
+
+    func run() throws {
+        let service = try makeService()
+
+        let options = ListCertificatesOptions(
+            filterSerial: filterSerial,
+            sort: sort,
+            filterType: filterType,
+            filterDisplayName: filterDisplayName,
+            limit: limit
+        )
+
+        let certificates = try service
+            .listCertificates(with: options)
+            .await()
+
+        if let downloadPath = downloadPath {
+            try certificates.forEach { (certificate: Certificate) in
+                guard let content = certificate.content else {
+                    throw CommandError.invalidContent
+                }
+
+                do {
+                    let folder = try Folder(path: downloadPath)
+
+                    let file = try folder.createFile(
+                        named: "\(certificate.serialNumber ?? "serial").cer",
```

Is this the expected default. Or should we throw an error if we don't get a name?

---

Or serial sorry

---

the most important thing about the certificate is its content. I don't quite sure if a certificate can only have `content`, without a `serialNumber `. What do you think?

---

I'm not sure, let's progress as is but we should think about declaring what we think is non optional and throw errors when we don't get those things in the future